### DISCUSSION TITLE
Test graphql subscriptions with Artillery and Appsync

### DISCUSSION THREAD

**Discussion by kyalha** [artilleryio/artillery#1207] [Artillery]
Hi there !

I am trying to find a way to test subscriptions with artillery with my appsync api.
In order to do so, I do this in my yaml file:
` - post:
          url: '/'
          json:
            query: | 
              subscription OnCreateMessage {
                  onCreateMessage {
                    id
                    userId
                    user {
                      id
                      name
                      messages {
                        nextToken
                      }
                      createdAt
                      updatedAt
                    }
                    content
                    createdAt
                    updatedAt
                  }
                }`
Unfortunately this gives me an error:
`  http:response "{\n  \"errors\" : [ {\n    \"errorType\" : \"BadRequestException\",\n    \"message\" : \"Subscriptions over MQTT is not supported.\"\n  } ]\n}" +0ms`

Has anyone figured out how to test a subscription with Appsync via Artillery ?
Mutations and queries work perfectly in my case, just the subscription where I don't see how to do it...

Thank you all !

**Comment by stanyq4** [comment]
@kyalha Check this out: https://github.com/z0heir/artillery-engine-graphql-ws#readme

**Comment by kyalha** [comment]
@stanyq4  thanks for your reply !
